testsuite/test_serverclass.py

```diff
@@ -0,0 +1,267 @@
+                      'Name': ns,
+                      'CreationClassName': 'CIM_Namespace',
+                      'ObjectManagerCreationClassName': 'CIM_ObjectManager',
+                      'SystemCreationClassName': 'CIM_ComputerSystem'}
+
+            ominst = self.inst_from_class(conn, "CIM_Namespace",
+                                          namespace=namespace,
+                                          PropertyValues=nsdict, strict=True,
+                                          IncludeNullProperties=False,
+                                          IncludePath=True)
+            conn.add_cimobjects(ominst, namespace=namespace)
+
+        namespaces = conn.EnumerateInstances("CIM_Namespace",
+                                             namespace=namespace)
+        assert len(namespaces) == len(test_namespaces)
+
+        # Build test CIM_RegisteredProfile instances
+        profiles = [('DMTF', 'Indications', '1.1.0'),
+                    ('DMTF', 'Profile Registration', '1.0.0'),
+                    ('SNIA', 'Server', '1.2.0')]
+        org_vm = ValueMapping.for_property(conn, namespace,
+                                           'CIM_RegisteredProfile',
+                                           'RegisteredOrganization')
+        # Map ValueMap to Value
+        org_vm_dict = {}
+        for value in range(0, 22):
+            org_vm_dict[org_vm.tovalues(value)] = value
+        for p in profiles:
+            instance_id = '%s+%s+%s' % (p[0], p[1], p[2])
+            reg_prof_dict = {'RegisteredOrganization': org_vm_dict[p[0]],
+                             'RegisteredName': p[1],
+                             'RegisteredVersion': p[2],
+                             'InstanceID': instance_id}
+            rpinst = self.inst_from_class(conn, "CIM_RegisteredProfile",
+                                          namespace=namespace,
+                                          PropertyValues=reg_prof_dict,
+                                          strict=True,
+                                          IncludeNullProperties=False,
+                                          IncludePath=True)
+
+            conn.add_cimobjects(rpinst, namespace=namespace)
+
+        assert(conn.EnumerateInstances("CIM_RegisteredProfile",
+                                       namespace=namespace))
+
+    @pytest.mark.parametrize(
+        "tst_namespace",
+        ['interop', 'root/interop', 'root/PG_Interop'])
+    def test_server_basic(self, tst_namespace):
+        """
+        Test the basic functions that access server information
+        """
+        conn = FakedWBEMConnection()
+        system_name = '"Mock Test_server_Class"'
+
+        self.build_instances(conn, tst_namespace, system_name)
+
+        server = WBEMServer(conn)
+
+        for iname in conn.EnumerateInstanceNames("CIM_Namespace",
+                                                 namespace=tst_namespace):
+            print('CIM_Namespace name %s' % iname)
```

Comment:
The test should be silent if it succeeds.

Reply:
Agreed. Done